### PR TITLE
fix(Tearsheet): description title show full text when overflow

### PR DESCRIPTION
**Pull request by Ratheeshrajan**
Contributes to #4230 

If customize column tearsheet `Description`  overflows two lines add `title` attribute to show the full text.

#### What did you change?
Added title to tearsheet description.
`src/components/Tearsheet/TearsheetShell.js`
image

#### How did you test and verify your work?
Storybook

### CODE REVIEW

**Comment by elycheea** [comment]
Been doing some reading on tooltip v. overflow thanks to recent discussions on them in filter panel (@paul-balchin-ibm @davidmenendez) and accessibility audits that @kristastarr have been in. Tooltip is probably the better approach for this upon further reading[^1][^2]. Sorry for the rework, but I’ll also update the notes in the issue. 😓 

[^1]: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#accessibility_concerns
[^2]: https://www.24a11y.com/2017/the-trials-and-tribulations-of-the-title-attribute/

**Comment by Ratheeshrajan** [comment]
@elycheea title replaced with tooltip.

**Comment by davidmenendez** [comment]
@elycheea @Ratheeshrajan i posted a [comment in the issue](https://github.com/carbon-design-system/ibm-products/issues/4230#issuecomment-2018735907) but i missed some of discussion going on here.

basically my opinion is that from an accessibility and UX standpoint that i don't think there's any reason to limit the text here and hide it through means of tooltip or title or anything.

i think the code looks great and it's a great solution if tooltip is the route we decide to take. @elycheea if you have any strong feelings one way or the other let me know!

File: packages/ibm-products/src/components/Tearsheet/TearsheetShell.js [hunk 1]
```diff
@@ -115,6 +116,23 @@ export const TearsheetShell = React.forwardRef(
...
+    const tooltipTriggerRef = useRef(null);
+    const [tooltipText, setTooltipText] = useState(null);
+
+    useEffect(() => {
+      if (open) {
+        if (
+          tooltipTriggerRef?.current?.scrollHeight >
+          tooltipTriggerRef?.current?.clientHeight
+        ) {
+          setTooltipText(tooltipTriggerRef?.current?.innerHTML);
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/Tearsheet/TearsheetShell.js:129]: i don't think it's necessary to set the text here. you should just be able to use the `description` prop in the tooltip label. i would probably just change this to be a boolean like `useTooltip` instead.

**Review by davidmenendez — CHANGES_REQUESTED** [review]
one comment

File: packages/ibm-products/src/components/Tearsheet/TearsheetShell.js (hunk 1 above)
Comment (elycheea) [reply packages/ibm-products/src/components/Tearsheet/TearsheetShell.js:129]: This is actually a lot like StringFormatter now. I think we should figure out how we want to handle utilities though.

**Review by davidmenendez — APPROVED** [review]

**Comment by davidmenendez** [comment]
@Ratheeshrajan please review the merge conflicts when you have time 👍

**Comment by elycheea** [comment]
Closing this for now and will move this issue back to backlog/triage for now. 

@Ratheeshrajan you’re welcome to reopen any of these PRs later if you’d like to address the conflicts — they’re probably quite a bit since we migrated several components to Typescript since this initial PR was opened. Also want to close these and move the issues back to the backlog in the meantime since you’re also on other projects now too. :)

**Pull request closed without merging**